docs/migration_guides/opflow_migration.rst

```diff
@@ -0,0 +1,1156 @@
+The :mod:`~qiskit.opflow.state_fns` module can be generally replaced by :class:`~qiskit.quantum_info.QuantumState`,
+with some differences to keep in mind:
+
+1. The primitives-based workflow does not rely on constructing state functions as opflow did
+2. Algorithm-specific functionality has been migrated to the respective algorithm's module
+
+
+Similarly to :class:`~qiskit.opflow.primitive_ops.PrimitiveOp`, :class:`~qiskit.opflow.state_fns.StateFn`
+acts as a factory to create the corresponding sub-class depending on the computational primitive used to initialize it.
+
+.. tip::
+
+    Interpreting :class:`~qiskit.opflow.state_fns.StateFn` as a factory class:
+
+    .. list-table::
+       :header-rows: 1
+
+       * - Class passed to :class:`~qiskit.opflow.state_fns.StateFn`
+         - Sub-class returned
+
+       * - ``str``, ``dict``, :class:`~qiskit.result.Result`
+         - :class:`~qiskit.opflow.state_fns.DictStateFn`
+
+       * - ``list``, ``np.ndarray``, :class:`~qiskit.quantum_info.Statevector`
+         - :class:`~qiskit.opflow.state_fns.VectorStateFn`
+
+       * - :class:`~qiskit.circuit.QuantumCircuit`, :class:`~qiskit.circuit.Instruction`
+         - :class:`~qiskit.opflow.state_fns.CircuitStateFn`
+
+       * - :class:`~qiskit.opflow.OperatorBase`
+         - :class:`~qiskit.opflow.state_fns.OperatorStateFn`
+
+This means that references to :class:`~qiskit.opflow.state_fns.StateFn` in opflow code should be examined to
+identify the sub-class that is being used, to then look for an alternative.
+
+.. list-table::
+   :header-rows: 1
+
+   * - Opflow
+     - Alternative
+
+   * - :class:`~qiskit.opflow.state_fns.StateFn`
+     - In most cases, :class:`~qiskit.quantum_info.Statevector`. Remember that this is a factory class.
+
+   * - :class:`~qiskit.opflow.state_fns.CircuitStateFn`
+     - :class:`~qiskit.quantum_info.Statevector`
+
+   * - :class:`~qiskit.opflow.state_fns.DictStateFn`
+     - :class:`~qiskit.quantum_info.Statevector`
+
+   * - :class:`~qiskit.opflow.state_fns.VectorStateFn`
+     - This class can be replaced with :class:`~qiskit.quantum_info.Statevector` or
+       :class:`~qiskit.quantum_info.StabilizerState` (for Clifford-based vectors).
+
+   * - :class:`~qiskit.opflow.state_fns.SparseVectorStateFn`
+     - No direct replacement.
+
+   * - :class:`~qiskit.opflow.state_fns.OperatorStateFn`
+     - :class:`~qiskit.quantum_info.Statevector`
+
+   * - :class:`~qiskit.opflow.state_fns.CVaRMeasurement`
+     - Used in :class:`~qiskit.opflow.expectations.CVaRExpectation`.
+       Functionality now covered by :class:`~SamplingEstimator`. See :ref:`example <cvar>` in expectations.
+
+
+Example 1: Applying an operator to a state
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+.. raw:: html
+
+    <details>
+    <summary><a><b>Opflow</b></a></summary>
+
+.. code-block:: python
+
+    from qiskit.opflow import StateFn, X, Y
+
+    qc = QuantumCircuit(2)
+    op = X ^ Y
+    state = StateFn(qc)
+
+    comp = ~state @ op
```

Comment:
This implements `<state | op` -- would `op | state>` be more intuitive to use? 😄 The latter is also what `Statevector.evolve` implements, I believe.

Reply:
Yes, I will change it.